infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java

```diff
@@ -258,4 +274,70 @@ private PersistentVolumeClaim replacePVCsWithCommon(
         .filter(subpath -> !isNullOrEmpty(subpath))
         .collect(Collectors.toSet());
   }
+
+  private void deleteCommonPVC(Workspace workspace) throws InfrastructureException {
+    factory.get(workspace).persistentVolumeClaims().delete(configuredPVCName);
+  }
+
+  /**
+   * @return true, if the common PVC is expected to be deleted, false otherwise. Depending on the
+   *     configuration it could be either the common PVC of a particular user, or the common PVC
+   *     that is shared across all the users (the last setup is considered to be uncommon and
+   *     not-recommended)
+   */
+  private boolean noWorkspacesLeft() {
+    return defaultNamespaceWithoutPlaceholderIsDefined()
+        ? totalNumberOfWorkspacesIsZero()
+        : userHasNoWorkspaces();
+  }
+
+  /**
+   * The method is expected to be used in order to identify if the common PVC is used across all the
+   * users. The common PVC for all the users will be used if
+   * 'che.infra.kubernetes.namespace.default' points to a particular namespace e.g. 'che',
+   * 'workspaces' etc.
+   *
+   * @return true, if 'che.infra.kubernetes.namespace.default' is defined and does NOT contain the
+   *     placeholder e.g.: che-workspace-<username>), false otherwise
+   */
```

Comment:
is this code still valid? Since now we have only per-user namespace strategy and <username> or <userid> placeholder is mandatory.

Reply:
@skabashnyuk nice catch, it looks like we do not need this check anymore - https://github.com/eclipse/che/pull/19409/files

Reply:
Thank you for the feedback, I removed the method and the two tests for this check